### PR TITLE
Sensors@claudiux: add support for Nvidia SMI temperatures

### PR DESCRIPTION
**Pull request by JEleniel**
Use the nvidia-smi tool (https://developer.nvidia.com/system-management-interface) to read GPU temperatures 

Spice author: @claudiux 
References: #6078

### CODE REVIEW

File: Sensors@claudiux/files/Sensors@claudiux/applet.js [hunk 1]
```diff
@@ -395,11 +396,13 @@ class SensorsApplet extends Applet.TextApplet {
         this.strictly_positive_fan ? 1 : 0,
         this.strictly_positive_volt ? 1 : 0
       );
+      this.reaper.reap_nvidia_smi();
     } else {
       this.set_applet_label(_("Suspended"));
     }
 
     this.loopId = Mainloop.timeout_add(this.interval * 1000, () => this.reap_sensors());
+    this.nvidiaLoopId = Mainloop.timeout_add(this.interval * 1000, () => this.reap_nvidia_smi());
```
Comment (claudiux) [inline Sensors@claudiux/files/Sensors@claudiux/applet.js:405]: The line 405 seems useless and contains an error.
`this.reap_nvidia_smi()` sould be `this.reaper.reap_nvidia_smi()` and a loop already exists calling `this.reap_sensors()` which calls `this.reaper.reap_nvidia_smi()`.

Please remove this.nvidiaLoopId from this applet.

**Review by claudiux — CHANGES_REQUESTED** [review]

**Comment by claudiux** [comment]
@JEleniel Thanks for the care. However, some changes are necessary.

**Review by claudiux — APPROVED** [review]

**Comment by FibroJedi** [comment]
@claudiux  At the risk of seeming dumb, have these changes gone out to the world? Only asking because my version of the applet isn't pulling in the NVIDIA temperatures per nvidia-smi or I can't find the option for it. I'm on 3.8.2 per the About dialog. (I didn't want to create a new issue in case there was no issue. Thanks!

**Comment by claudiux** [comment]
Ping @JEleniel

**Comment by JEleniel** [comment]
Hello @FibroJedi, I am the author of the Nvidia code. The changes are in 3.8.2. 

The *nvidia-smi* tool is part of the Nvidia closed source drivers, only supports ["NVIDIA's Tesla, Quadro, GRID and GeForce devices from Fermi and higher architecture families"](https://docs.nvidia.com/deploy/nvidia-smi/index.html), and the applet requires version 550.107.02 or higher.

If you have a compatible GPU and the closed source drivers installed, please post the output of `nvidia-smi -L`. You can also try the `Windows-L` key combination to bring up the Cinnamon debugger (melange) or look in the `~/.xsession-errors` log for errors - search for "Nvidia SMI".

**Comment by FibroJedi** [comment]
> Hello @FibroJedi, I am the author of the Nvidia code. The changes are in 3.8.2.
> 
> The _nvidia-smi_ tool is part of the Nvidia closed source drivers, only supports ["NVIDIA's Tesla, Quadro, GRID and GeForce devices from Fermi and higher architecture families"](https://docs.nvidia.com/deploy/nvidia-smi/index.html), and the applet requires version 550.107.02 or higher.
> 
> If you have a compatible GPU and the closed source drivers installed, please post the output of `nvidia-smi -L`. You can also try the `Windows-L` key combination to bring up the Cinnamon debugger (melange) or look in the `~/.xsession-errors` log for errors - search for "Nvidia SMI".

I have closed source drivers and nvidia-smi installed (and works). Result of `nvidia-smi -L`
> GPU 0: NVIDIA GeForce RTX 3050

But I'm currently on version 535 Drivers. I was going to upgrade them today anyway. I've just had issues in the past with upgrading the drivers, then the system not rebooting properly. I'll see if I can get up to 550 without incident then check the applet.

Thanks for the info - and sorry if that was somewhere in the documentation!

**Comment by FibroJedi** [comment]
Now I have, thankfully, successfully upgraded to driver-550, the NVIDIA option in Sensors is showing. Thank you!

**Comment by JEleniel** [comment]
I'm glad you were able to get it working. I should add that information to the documentation.

**Pull request merged**